src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportOrchestratorJob.cs

```diff
@@ -0,0 +1,160 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Features.Operations.Export.Models;
+using Microsoft.Health.Fhir.Core.Features.Search;
+using Microsoft.Health.Fhir.Core.Models;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Export
+{
+    [JobTypeId((int)JobType.ExportOrchestrator)]
+    public class ExportOrchestratorJob : IJob
+    {
+        private const int DefaultNumberOfSurrogateIdRanges = 100;
+
+        private IQueueClient _queueClient;
+        private ISearchService _searchService;
+        private ILogger<ExportOrchestratorJob> _logger; // TODO: Either remove or use
+
+        public ExportOrchestratorJob(
+            IQueueClient queueClient,
+            ISearchService searchService,
+            ILoggerFactory loggerFactory)
+        {
+            EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            EnsureArg.IsNotNull(searchService, nameof(searchService));
+            EnsureArg.IsNotNull(loggerFactory, nameof(loggerFactory));
+
+            _queueClient = queueClient;
+            _searchService = searchService;
+            _logger = loggerFactory.CreateLogger<ExportOrchestratorJob>();
+        }
+
+        internal int NumberOfSurrogateIdRanges { get; set; } = DefaultNumberOfSurrogateIdRanges;
+
+        public async Task<string> ExecuteAsync(JobInfo jobInfo, IProgress<string> progress, CancellationToken cancellationToken)
+        {
+            var record = JsonConvert.DeserializeObject<ExportJobRecord>(jobInfo.Definition);
```

Comment:
Missing EnsureArg.IsNotNull(jobInfo, nameof(jobInfo)); and EnsureArg.IsNotNull(progress, nameof(progress)); ?